src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -126,26 +130,36 @@ impl RowSetIterator {
         // TODO: parallel fetch
         // TODO: align unmatched rows
 
-        for id in 0..filter_column.len() {
-            if filter_column[id] {
-                if let Some((row_id, array)) = self.column_iterators[id]
-                    .as_mut()
-                    .unwrap()
-                    .next_batch(Some(fetch_size), None)
-                    .await?
-                {
-                    if let Some(x) = common_chunk_range {
-                        if x != (row_id, array.len()) {
-                            panic!("unmatched rowid from column iterator");
+        for (id, column_ref) in self.column_refs.iter().enumerate() {
+            if has_filter {
```

Comment:
I would prefer using `if let Some(filter_context) = filter_context`, instead of combining `.is_some()` with `unwrap()`.